frontstage/views/passwords.py

```diff
@@ -92,32 +70,36 @@ def forgot_password_check_email():
 @passwords_bp.route('/reset-password/<token>', methods=['GET', 'POST'])
 def reset_password(token):
     form = ResetPasswordForm(request.form)
-    url = app.config['RAS_PARTY_VERIFY_PASSWORD_TOKEN'].format(app.config['RAS_PARTY_SERVICE'], token)
-    response = requests.get(url, auth=app.config['BASIC_AUTH'], verify=False)
-    if response.status_code == 409:
-        logger.warning('Token expired', token=token)
-        return render_template('passwords/password-expired.html', _theme='default')
-    elif response.status_code == 404:
-        logger.warning('Invalid token sent to party service', token=token)
-        return redirect(url_for('error_bp.not_found_error_page'))
-    elif response.status_code != 200:
-        logger.error('Party service failed to verify token')
-        raise ExternalServiceError(response)
 
     if request.method == 'POST' and form.validate():
         password = request.form.get('password')
         put_data = {
-            "new_password": password
+            "new_password": password,
+            "token": token
         }
 
-        url = app.config['RAS_PARTY_CHANGE_PASSWORD'].format(app.config['RAS_PARTY_SERVICE'], token)
-        response = requests.put(url, auth=app.config['BASIC_AUTH'], json=put_data, verify=False)
+        url = app.config['CHANGE_PASSWORD']
+        response = api_call('PUT', url, json=put_data)
 
-        if response.status_code != 200:
+        if response.status_code == 200:
+            logger.info('Successfully change user password', token=token)
+            return redirect(url_for('passwords_bp.reset_password_confirmation'))
+        else:
             logger.error('Failed to change user password', token=token)
-            raise ExternalServiceError(response)
-        logger.info('Successfully change user password', token=token)
-        return redirect(url_for('passwords_bp.reset_password_confirmation'))
+    else:
+        url = app.config['VERIFY_PASSWORD_TOKEN']
+        parameters = {"token": token}
+        response = api_call('GET', url, parameters=parameters)
+
+    if response.status_code == 409:
+        logger.warning('Token expired', token=token)
+        return render_template('passwords/password-expired.html', _theme='default')
+    elif response.status_code == 404:
+        logger.warning('Invalid token sent to party service', token=token)
+        return redirect(url_for('error_bp.not_found_error_page'))
+    elif response.status_code != 200:
+        logger.error('Party service failed to verify token')
```

Comment:
If we logged the stacktrace I guess this line of logging would be unnecessary and would caused duplicate error logging as we will log when the ApiError is raised?

Reply:
This is being done throughout front stage I'll raise a ticket to make these changes throughout to keep the service consistent.